test/jest/victory-bar/victory-bar.spec.js

```diff
@@ -0,0 +1,257 @@
+        return getShapeUtils.getBarHeight(commandString);
+      });
+
+      expect(Math.trunc(heights[1] / 2)).toEqual(Math.trunc(heights[0], 0.5));
+      expect(((heights[2] / 3) * 2).toFixed(3)).toEqual(heights[1].toFixed(3));
+    });
+
+    it("does not render data with null x or y values", () => {
+      const data = [
+        { x: 1, y: 2 },
+        { x: null, y: 4 },
+        { x: 5, y: null }
+      ];
+      const { container } = render(<VictoryBar data={data} />);
+      expect(container.querySelectorAll("path")).toHaveLength(1);
+    });
+  });
+
+  describe("event handling", () => {
+    const clickHandler = jest.fn();
+
+    beforeEach(() => {
+      clickHandler.mockReset();
+    });
+
+    it("attaches an event to the parent svg", () => {
+      const { container } = render(
+        <VictoryBar
+          events={[
+            {
+              target: "parent",
+              eventHandlers: { onClick: clickHandler }
+            }
+          ]}
+        />
+      );
+      const bar = container.querySelector("path");
+      fireEvent.click(bar);
+
+      expect(clickHandler).toHaveBeenCalled();
+    });
+
+    it("attaches an event to data", () => {
+      const data = [
+        { x: 0, y: 0, label: "0" },
+        { x: 1, y: 1, label: "1" },
+        { x: 2, y: 2, label: "2" }
+      ];
+      const { container } = render(
+        <VictoryBar
+          data={data}
+          events={[
+            {
+              target: "data",
+              eventHandlers: { onClick: clickHandler }
+            }
+          ]}
+        />
+      );
+      const bars = container.querySelectorAll("path");
+      bars.forEach((bar, index) => {
+        clickHandler.mockReset();
+        fireEvent.click(bar);
+        expect(clickHandler).toHaveBeenCalled();
+
+        const barContext = clickHandler.mock.calls[0][1];
+        expect(barContext.datum.x).toEqual(data[index].x);
+      });
+    });
+
+    it("attaches an event to a label", () => {
+      const data = [
+        { x: 0, y: 0, label: "label 0" },
+        { x: 1, y: 1, label: "label 1" },
+        { x: 2, y: 2, label: "label 2" }
+      ];
+
+      render(
+        <VictoryBar
+          data={data}
+          labels={({ datum }) => datum.label}
+          events={[
+            {
+              target: "labels",
+              eventHandlers: { onClick: clickHandler }
+            }
+          ]}
+        />
+      );
+      const label = screen.getByText("label 1");
+
+      fireEvent.click(label);
+
+      expect(clickHandler).toHaveBeenCalled();
+    });
+  });
+
+  describe("accessibility", () => {
+    it("adds an aria role to each bar in the series", () => {
+      const data = range(20).map((y, x) => ({ x, y }));
+      const { container } = render(<VictoryBar data={data} />);
+
+      Array.from(container.querySelectorAll("path")).forEach((bar) => {
+        expect(bar.getAttribute("role")).toEqual("presentation");
```

Comment:
Rather than iterating through each bar, you could do something like `expect(screen.queryByRole('presentation')).toHaveLength(20)`